### PR TITLE
Rename MSSE into RelMSE, add new implementation of MSSE

### PR DESCRIPTION
**Pull request by nickto**
Following the discussion in #184:

- Rename `msse()` into `rel_msse()` (MSSE into RelMSE).
- Add MSSE implementation, new `msse()` function, more consistent with [Hyndman and Koehler (2006)](https://www.sciencedirect.com/science/article/pii/S0169207006000239) and [Makridakis et al. (2022)](https://www.sciencedirect.com/science/article/pii/S0169207021001874).

### CODE REVIEW

File: hierarchicalforecast/evaluation.py [hunk 1]
```diff
@@ -145,6 +145,50 @@ def msse(y, y_hat, y_train, mask=None):
...
 
 # %% ../nbs/evaluation.ipynb 11
+def msse(y, y_hat, y_train, mask=None):
+    """Mean Squared Scaled Error
+
+    Computes Mean squared scaled error (MSSE), as proposed by Hyndman & Koehler (2006)
+    as an alternative to percentage errors, to avoid measure unstability.
+
+    $$ \\mathrm{MSSE}(\\mathbf{y}, \\mathbf{\\hat{y}}, \\mathbf{y}^{in-sample}) =
+    \\frac{\\frac{1}{h} \\sum^{n+h}_{t=n+1} (y_t - \\hat{y}_t)^2}{\\frac{1}{n-1} \\sum^{n}_{t=2} (y_t - y_{t-1})^2},$$
```
Comment (kdgutier) [inline hierarchicalforecast/evaluation.py:155]: Can we change this equation:
$$ \\mathrm{MSSE}(\\mathbf{y}, \\mathbf{\\hat{y}}, \\mathbf{y}^{in-sample}) =
  \\frac{\\frac{1}{h} \\sum^{n+h}_{t=n+1} (y_t - \\hat{y}_t)^2}{\\frac{1}{n-1} \\sum^{n}_{t=2} (y_t - y_{t-1})^2},$$

To match Nixtla's losses notation:
$$ \\mathrm{MSSE}(\\mathbf{y}, \\mathbf{\\hat{y}}, \\mathbf{y}^{in-sample}) =
  \\frac{\\frac{1}{h} \\sum^{t+h}_{\\tau=t+1} (y_{\\tau} - \\hat{y}_{\\tau})^2}{\\frac{1}{t-1} \\sum^{n}_{\\tau=2} (y_{\\tau} - y_{\\tau-1})^2},$$

File: hierarchicalforecast/evaluation.py [hunk 2]
```diff
@@ -145,6 +145,50 @@ def msse(y, y_hat, y_train, mask=None):
...
+    `mask`: numpy array, Specifies date stamps per serie to consider in loss.<br>
+
+    **Returns:**<br>
+    `loss`: float.    
+
+    **References:**<br>
+    - [Hyndman, R. J and Koehler, A. B. (2006).
+       "Another look at measures of forecast accuracy",
+       International Journal of Forecasting, Volume 22, Issue 4.](https://www.sciencedirect.com/science/article/pii/S0169207006000239)<br>
+    - [Kin G. Olivares, O. Nganba Meetei, Ruijun Ma, Rohan Reddy, Mengfei Cao, Lee Dicker.
```
Comment (kdgutier) [inline hierarchicalforecast/evaluation.py:172]: Let's keep only Hyndman (2006) reference for the MSSE

**Review by kdgutier — COMMENTED** [review]
Pretty cool @nickto,

Thanks for adding the `msse` and renaming `mse` into `rel_mse`.
I added some comments regarding the documentation and msse reference.

File: hierarchicalforecast/evaluation.py (hunk 1 above)
Comment (nickto) [reply hierarchicalforecast/evaluation.py]: Updated with some minor changes.

**Review by kdgutier — APPROVED** [review]

**Review by kdgutier — APPROVED** [review]

**Pull request merged**